java/README.rst

```diff
@@ -0,0 +1,254 @@
+- ``withVerbosity()``: Logs to the server span additional events, such as message received, half close (client finished sending messages), and call complete. Default only logs if a call is cancelled.  
+- ``withTracedAttributes(ServerRequestAttribute... attrs)``: Sets tags on the server span in case you want to track information about the RPC call. See ServerRequestAttribute.java for a list of traceable request attributes.
+
+**Example**:
+
+.. code-block:: java
+
+    ServerTracingInterceptor tracingInterceptor = new ServerTracingInterceptor
+        .Builder(tracer)
+        .withStreaming()
+        .withVerbosity()
+        .withOperationName("service-name")
+        .withTracedAttributes(ServerRequestAttribute.HEADERS, 
+            ServerRequestAttribute.METHOD_TYPE)
+        .build();
+
+==============
+Client Tracing
+==============
+
+A ``ClientTracingInterceptor`` also has default settings, which you can override by creating it using a ``ServerTracingInterceptor.Builder``.
+
+- ``withOperationName(String operationName)``: Define how the operation name is constructed for all spans created for this intercepted client. Default is the name of the RPC method. More details `below`_.
+- ``withStreaming()``: Logs to the client span whenever a message is sent or a response is received.
+- ``withVerbosity()``: Logs to the client span additional events, such as call started, message sent, half close (client finished sending messages), response received, and call complete. Default only logs if a call is cancelled.  
+- ``withTracedAttributes(ClientRequestAttribute... attrs)``: Sets tags on the client span in case you want to track information about the RPC call. See ClientRequestAttribute.java for a list of traceable request attributes.
+
+**Example**:
+
+.. code-block:: java
+
+    ClientTracingInterceptor tracingInterceptor = new ClientTracingInterceptor
+        .Builder(tracer)
+        .withStreaming()
+        .withVerbosity()
+        .withOperationName("client-name")
+        .withTracingAttributes(ClientRequestAttribute.ALL_CALL_OPTIONS,
+            ClientRequestAttribute.HEADERS)
+        .build();
+
+.. _below: `Operation Names`_
+
+====================
+Current Span Context
+====================
+
+In your server request handler, you can access the current active span for that request by calling
+
+.. code-block:: java
+
+    Span span = OpenTracingContextKey.activeSpan();
+
+This is useful if you want to manually set tags on the span, log important events, or create a new child span for internal units of work. You can also use this key to wrap these internal units of work with a new context that has a user-defined active span.
+
+For example:
+
+.. code-block:: java
+
+    Tracer tracer = aTracerImpl;
+
+    // some unit of internal work that you want to trace
+    Runnable internalWork = someInternalWork
+
+    // a wrapper that traces the work of the runnable
+    class TracedRunnable implements Runnable {
```

Comment:
@bensigelman not sure if this example is any more clear than before; let me know if you had something different in mind or if this is ok